Styling & Weights Page

Josh-Padua opened the pull request:
## 📋 Description
<!-- What does this PR do? Briefly summarize the changes. -->
- Most changes were styling and making the application look cohesive and fit together. Other functional features include: graph now reads live data, weights tracking now implemented + editor functional as well


## ✅ Changes Made
<!-- List the key changes in this PR. -->
- Weight tracker + editor page added.
- View padding fixed so all buttons will be visible no matter how far you scroll
- Nativewind styling added to stopwatch
- Nativewind styling added to journal 
- Macro graph on dashboard now takes real data from the database and displays it

## 🧠 Reason for Change
<!-- Why were these changes necessary? What problem does it solve? -->
- Creates a more uniform feel to the UI of the application
- Weights page to edit previous entries and view the previous entries to see the users progress

## 🧪 How to Test
<!-- Steps to test the changes locally. -->
1. Enter an update weight on the Dashboard go to the weights page (button below update) to view weight entries + edit previous entries. Check database to make sure changes were saved 

## ⚠️ Notes
<!-- Optional: any caveats, follow-up tasks, or extra context. -->
- Still need to try read the data from the database to display Weight-over-time graph. at the moment it is hard coded as I've had difficulty finding a way to display it properly.

Caleb-Pace commented:
This is more of a side note, for a future fix. But you aren't updating the data on the profile page (the DB is updated), it only updates when the app is re-opened.

Another thing is, there are 2 'Update Details' buttons and they both trigger and update of the first and last name. This isn't too important but it does make there being 2 buttons redundant.

You could also clear the text input once the details have been updated. This would be a good contender for a UX update (to help it flow better).

Caleb-Pace commented:
I understand why you have done the view weight logs like that but it could definitely be improved upon for better UX.

Caleb-Pace commented:
The 'Lap' button for the stopwatch page might benefit from starting greyed out for better HCI.

Caleb-Pace reviewed (changes requested):
Home page and Profile page are not updating data. (I believe they only query from when the app starts and do not update this).
To fix this look into events or how the Macros page does this.

Weight does not update (I know you mentioned this in your notes section of the PR but this appears to be the main feature of your PR).
You can take the last 5 or 7 weight entries and graph those. (Remember to update this when the 'Update Weight' button is pressed).

Josh-Padua commented:
> This is more of a side note, for a future fix. But you aren't updating the data on the profile page (the DB is updated), it only updates when the app is re-opened.
> 
> Another thing is, there are 2 'Update Details' buttons and they both trigger and update of the first and last name. This isn't too important but it does make there being 2 buttons redundant.
> 
> You could also clear the text input once the details have been updated. This would be a good contender for a UX update (to help it flow better).

one update details button removed. was originally there for testing but I updated the way that handleUpdate updates the DB and it now only updates the fields that have inputs so only 1 button is required.

Text input fields now changed so that handleUpdate sets setFirstName and setLastName to empty strings and is displayed as follows on the settings page.

Data updates on profile page and dashboard live. utilizes useFocusEffect which I found from react and used AI to implement into React Native, so that when the page is focused it refetches newly updated data from different pages (i.e. macros + settings)

Josh-Padua commented:
> The 'Lap' button for the stopwatch page might benefit from starting greyed out for better HCI.

Lap button when not running is now grey. kept the same color scheme when running. also changed the start and stop for better UI and now goes green or red.

Josh-Padua commented:
> Home page and Profile page are not updating data. (I believe they only query from when the app starts and do not update this). To fix this look into events or how the Macros page does this.
> 
> Weight does not update (I know you mentioned this in your notes section of the PR but this appears to be the main feature of your PR). You can take the last 5 or 7 weight entries and graph those. (Remember to update this when the 'Update Weight' button is pressed).
> 
> Also if you can set a max-width for the statistics on the profile page

max-width set for the profile page.
weights graph still not functional. I've tried different ways but I will try some other approaches later today.
dashboard + profile now update data.

Caleb-Pace reviewed (changes requested):
Still no maximum width for the profile statistics.

Weight graph does not update.

Josh-Padua commented:
> Still no maximum width for the profile statistics.
> 
> Weight graph does not update.

weight graph live updates. max width for profile as well as graphs changed for better fit.

Caleb-Pace reviewed (approved):
Good
